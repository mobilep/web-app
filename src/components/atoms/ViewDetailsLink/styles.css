.ViewDetailsLink {
	background: transparent;
	text-align: center;
	color: currentColor;
	border-color: currentColor;
}

.ViewDetailsLink:hover {
	color: currentColor;
	border-color: currentColor;
}
