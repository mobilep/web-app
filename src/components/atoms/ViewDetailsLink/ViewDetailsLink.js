import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Button from '../Button';
import './styles.css';

export default class ViewDetailsLink extends Component {

	static propTypes = {
		to: PropTypes.string.isRequired,
	}

	getClassName ({ className }) {
		return classNames('ViewDetailsLink', className);
	}

	render () {
		const className = this.getClassName(this.props);
		return <Button {...this.props} className={className} />;
	}

}
