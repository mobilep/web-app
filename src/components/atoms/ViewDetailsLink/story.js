import React from 'react';
import { storiesOf } from '@storybook/react';

import ViewDetailsLink from './ViewDetailsLink';

storiesOf('1. Atoms / ViewDetailsLink', module)
	.add('with text', () => <ViewDetailsLink>Yo</ViewDetailsLink>);
