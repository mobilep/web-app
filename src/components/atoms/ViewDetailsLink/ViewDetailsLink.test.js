import React from 'react';
import { shallow } from 'enzyme';

import ViewDetailsLink from './ViewDetailsLink';

describe('Atom - ViewDetailsLink', () => {
	it(`Should render`, () => {
		const wrapper = shallow(
			<ViewDetailsLink to='someTestUrl' />
		);
		expect(wrapper).toMatchSnapshot();
	});

});
