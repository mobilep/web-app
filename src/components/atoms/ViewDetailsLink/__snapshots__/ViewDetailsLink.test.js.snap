// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - ViewDetailsLink Should render 1`] = `
<Button
  className="ViewDetailsLink"
  to="someTestUrl"
/>
`;
